java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,343 @@
+                        } else {
+                            dateMilliVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.TIME:
+                        Time time = rs.getTime(i);
+                        TimeMilliVector timeMilliVector = (TimeMilliVector)root.getVector(columnName);
+                        timeMilliVector.setValueCount(rowCount + 1);
+                        if (time != null) {
+                            timeMilliVector.setSafe(rowCount, (int) rs.getTime(i).getTime());
+                        } else {
+                            timeMilliVector.setNull(rowCount);
+                        }
+
+                        break;
+                    case Types.TIMESTAMP:
+                        // timezone is null
+                        Timestamp timestamp = rs.getTimestamp(i);
+                        TimeStampVector timeStampVector = (TimeStampVector)root.getVector(columnName);
+                        timeStampVector.setValueCount(rowCount + 1);
+                        if (timestamp != null) {
+                            timeStampVector.setSafe(rowCount, timestamp.getTime());
+                        } else {
+                            timeStampVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.BINARY:
+                    case Types.VARBINARY:
+                    case Types.LONGVARBINARY:
+                        VarBinaryVector varBinaryVector = (VarBinaryVector)root.getVector(columnName);;
+                        varBinaryVector.setValueCount(rowCount + 1);
+                        byte[] bytes = rs.getBytes(i);
+                        if (bytes != null) {
+                            varBinaryVector.setIndexDefined(rowCount);
+                            varBinaryVector.setValueLengthSafe(rowCount, bytes.length);
+                            varBinaryVector.setSafe(rowCount, bytes);
+                        } else {
+                            varBinaryVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.ARRAY:
+                        // not handled
+//                    fields.add(new Field("list", FieldType.nullable(new ArrowType.List()), null));
+                        break;
+                    case Types.CLOB:
+                        VarCharVector varcharVector1 = (VarCharVector)root.getVector(columnName);
+                        varcharVector1.setValueCount(rowCount + 1);
+                        Clob clob = rs.getClob(i);
+                        if (clob != null) {
+                            int length = (int) clob.length();
+                            varcharVector1.setIndexDefined(rowCount);
+                            varcharVector1.setValueLengthSafe(rowCount, length);
+                            varcharVector1.setSafe(rowCount, clob.getSubString(1, length).getBytes(), 0, length);
```

Comment:
- better use a streaming API
- an encoding should be specified for `getBytes()`

Reply:
What would be your suggestion for encoding? Should I take it as one of the API input parameters from the caller?

Reply:
UTF_8 is fine I think (since arrow varchar are UTF8 based). The issue is that getBytes() takes whatever the default encoding of the platform is...